src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        clearObject(stateCopy);
+
+        break;
+
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateCopy, action.keysToRemove);
+
+        break;
+
+      default:
+        continue;
```

Comment:
[CODE STYLE]: The `default` case should be used for error handling, but here it silently continues the loop. It would be better to handle unexpected `action.type` values more explicitly, either by logging a warning or throwing an error.